File: Memorize/TextEditorViewController.m
//
//  TextEditorViewController.m
//  Memorize
//
//  Created by Nick Shelley on 4/27/14.
//  Copyright (c) 2014 Mine. All rights reserved.
//

#import "TextEditorViewController.h"

@interface TextEditorViewController ()

@property (weak, nonatomic) IBOutlet UITextView *textView;

@end

@implementation TextEditorViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.textView.text = self.existingText ?: @"";
    [self.textView becomeFirstResponder];
}

- (IBAction)cancelPressed {
    [self dismissViewControllerAnimated:YES completion:nil];
}

- (IBAction)savePressed {
    if (self.saveBlock) {
        self.saveBlock(self.textView.text);
        self.saveBlock = nil;
    }
    [self dismissViewControllerAnimated:YES completion:nil];
}

@end

File: Memorize/TextEditorViewController.swift
//
//  TextEditorViewController.swift
//  Memorize
//
//  Created by Heather Shelley on 2/25/17.
//  Copyright © 2017 Mine. All rights reserved.
//

import UIKit

class TextEditorViewController: UIViewController {
    var existingText: String?
    var didSave: (String) -> Void = { _ in }
    
    @IBOutlet weak var textView: UITextView?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        textView?.text = self.existingText ?? ""
        textView?.becomeFirstResponder()
    }
    
    @IBAction func cancelPressed() {
        dismiss(animated: true, completion: { _ in })
    }
    
    @IBAction func savePressed() {
        didSave(textView?.text ?? "")
        dismiss(animated: true, completion: nil)
    }
}
